frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,625 @@
+          onClose={() => undefined}
+        />
+      )}
+    />,
+    {
+      withRouter: true,
+      storeInitialState: {
+        currentUser: CURRENT_USER,
+      },
+    },
+  );
+
+  if (waitForContent) {
+    await waitFor(() => {
+      expect(screen.queryByText("Loading...")).not.toBeInTheDocument();
+    });
+  }
+};
+
+describe("AddToDashSelectDashModal", () => {
+  describe("Create new Dashboard", () => {
+    it("should open CreateDashboardModal", async () => {
+      await setup({
+        mostRecentlyViewedDashboard: DASHBOARD,
+      });
+
+      const createNewDashboard = screen.getByRole("heading", {
+        name: "Create a new dashboard",
+      });
+
+      userEvent.click(createNewDashboard);
+
+      // opened CreateDashboardModal
+      expect(
+        screen.getByRole("heading", {
+          name: /new dashboard/i,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  describe("Add to existing Dashboard", () => {
+    it("should show loading", async () => {
+      await setup({
+        waitForContent: false,
+        mostRecentlyViewedDashboard: DASHBOARD,
+      });
+
+      expect(screen.getByText("Loading...")).toBeInTheDocument();
+    });
+
+    it("should show error", async () => {
+      const ERROR = "Server Error!";
+      await setup({ error: ERROR, mostRecentlyViewedDashboard: DASHBOARD });
+
+      expect(screen.getByText(ERROR)).toBeInTheDocument();
+    });
+
+    describe("when user visited some dashboard in last 24hrs", () => {
+      it("should preselected last visited dashboard in the picker", async () => {
+        await setup({
+          mostRecentlyViewedDashboard: DASHBOARD,
+        });
+
+        const dashboardCollection = checkNotNull(
+          COLLECTIONS.find(
+            collection => collection.id === DASHBOARD.collection_id,
+          ),
+        );
+
+        // breadcrumbs
+        assertBreadcrumbs([dashboardCollection]);
+        expect(screen.getByText(DASHBOARD.name)).toBeInTheDocument();
+      });
+
+      describe("when last visited dashboard belongs to root", () => {
+        it("should render root collection", async () => {
+          await setup({
+            dashboard: DASHBOARD_AT_ROOT,
+            mostRecentlyViewedDashboard: DASHBOARD_AT_ROOT,
+          });
+
+          // breadcrumbs
+          assertBreadcrumbs([ROOT_COLLECTION]);
+          expect(screen.getByText(DASHBOARD_AT_ROOT.name)).toBeInTheDocument();
+        });
+      });
+
+      describe("when last visited dashboard belongs to public subcollections", () => {
+        const dashboardInPublicSubcollection = createMockDashboard({
+          id: 5,
+          name: "Dashboard in public subcollection",
+          collection: SUBCOLLECTION,
+          collection_id: SUBCOLLECTION.id as number,
+          model: "dashboard",
+        });
+
+        it("should show most recently visited dashboard if the question is in a public collection", async () => {
```

Comment:
test Query builder > Newly created question > #7 from [the epic](https://github.com/metabase/metabase/issues/34733)